shiksha-api/app-service/app/services/rag_adapter_cache.py

```
@@ -1,190 +1,71 @@
+            await evicted.cleanup()
         return adapter
 
-    async def cleanup(self) -> None:
-        for adapter in self._rag_adapter_cache.values():
+
+    async def get(self, index_path: str, completion_llm: OpenAIResponses, embedding_llm: OpenAIEmbedding) -> T:
+        async with self._lock_for(index_path):
+            return await self._actually_get(index_path, completion_llm, embedding_llm)
+
+
+    async def _cleanup_one(self, index_path: str):
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `_cleanup_one`

Add return type annotation: `None`

(ANN202)